kernel/src/table_changes/resolve_dvs.rs

```diff
@@ -0,0 +1,420 @@
+    /// in the CDF output. Otherwise the row must be filtered out. The vector may be shorter than
+    /// the data file. In this case, all the remaining rows are *not* selected. If `selection_vector`
+    /// is `None`, then all rows are selected.
+    selection_vector: Option<Vec<bool>>,
+}
+
+/// Resolves the deletion vectors for a [`CdfScanFile`]. This function handles two
+/// types of `CdfScanFile`s:
+/// 1. The first case is a [`CdfScanFile`] paired with a remove deletion vector. The `scan_type`
+///    must be [`CdfScanFileType::Add`]. In this case, both the add and remove deletion vectors are
+///    read if they exist. Then, we find the set of rows in the that have been added, and  rows that
+///    have been removed. The set of removed rows (if any) will be represented by a
+///    [`ResolvedCdfScanFile`] with `scan_type` = [`CdfScanFileType::Remove`]. The set of added rows
+///    (if any) will be  represented by a [`ResolvedCdfScanFile`] with `scan_type` = [`CdfScanFileType::Add`].
+///
+///    Note: We allow the possibility for there to be both added rows and removed rows for a
+///    single add/remove pair.
+/// 2. The second case handles all other add, remove, and cdc [`CdfScanFile`]s. These will simply
+///    read the deletion vector (if present), and each is converted into a [`ResolvedCdfScanFile`].
+///    No changes are made to the `scan_type`.
+#[allow(unused)]
+fn resolve_scan_file_dv(
+    engine: &dyn Engine,
+    table_root: &Url,
+    scan_file: CdfScanFile,
+) -> DeltaResult<impl Iterator<Item = ResolvedCdfScanFile>> {
+    let add_dv = scan_file.dv_info.get_treemap(engine, table_root)?;
+    let rm_dv = scan_file
+        .remove_dv
+        .as_ref()
+        .map(|rm_dv| rm_dv.get_treemap(engine, table_root))
+        .transpose()?;
+    let (add_dv, rm_dv) = match (add_dv, rm_dv, &scan_file.scan_type) {
+        (_, Some(_), CdfScanFileType::Remove) => {
+            return Err(Error::generic(
+                "CdfScanFile with type remove cannot have a remove deletion vector",
+            ));
+        }
+        (_, Some(_), CdfScanFileType::Cdc) => {
+            return Err(Error::generic(
+                "CdfScanFile with type cdc cannot have a remove deletion vector",
+            ));
+        }
+        (add_dv, Some(rm_dv), CdfScanFileType::Add) => {
+            let add_dv = add_dv.unwrap_or_else(Default::default);
+            let rm_dv = rm_dv.unwrap_or_else(Default::default);
+            // Here we show how deletion vectors are resolved. Note that logically the `rm_dv` is the
+            // beginning state of the commit, and `add_dv` is the final state of the commit. In
+            // other words the dv went from being `rm_dv` to `add_dv`.
+            //
+            // ===== IMPORTANT =====
+            // It is important to note that `rm_dv` and `add_dv` are deletion treemaps. We define
+            // two type of treemaps:
```

Comment:
```suggestion
            // Both `rm_dv` and `add_dv` are deletion treemaps. We define two type of treemaps:
```
(already said it was important)